src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,95 @@
 package core.basesyntax;
 
+import static core.basesyntax.MyHashMap.Node.hash;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_LENGTH = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] hashTable = new Node[DEFAULT_LENGTH];
+    private int size;
+    private int arrayLength;
+
+    static class Node<K, V> implements Entry<K, V> {
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+
+        static int hash(Object key) {
+            int h;
+            return (key == null) ? 0 : (h = key.hashCode()) * (h >> 4);
+        }
+    }
+
+    public MyHashMap() {
+        this.arrayLength = DEFAULT_LENGTH;
+    }
 
     @Override
     public void put(K key, V value) {
+        int hashCode = hash(key);
+        int index = Math.abs(hashCode) % arrayLength;
```

Comment:
```suggestion
        int index = Math.abs(hashCode) % table.length;
```
and move it to a separate method, you are using it again